packages/synapse-react-client/src/components/TableColumnSchemaEditor/ColumnModelForm.tsx

```diff
@@ -289,8 +305,19 @@ export default function ColumnModelForm(props: ColumnModelFormProps) {
               })
             }}
             fullWidth
-            error={!!errorsByField['maximumSize']}
-            helperText={errorsByField['maximumSize']}
+            error={
+              recommendedSize && typeof currentMaxSize === 'number'
+                ? currentMaxSize < recommendedSize ||
+                  !!errorsByField['maximumSize']
+                : !!errorsByField['maximumSize']
+            }
+            helperText={
+              recommendedSize &&
+              typeof currentMaxSize === 'number' &&
+              currentMaxSize < recommendedSize
+                ? `Recommended size is at least ${recommendedSize}`
+                : errorsByField['maximumSize']
+            }
```

Comment:
This is a lot of complexity in this JSX, can you colocate this code with the `recommendedSize` code?

Reply:
The error should also have a higher priority to be shown than the new warning